generate_isoscape.py

```diff
@@ -0,0 +1,191 @@
+  """
+  dispatch_rasters function
+  --------------------------------------------------
+  Loads rasters either from Google Earth Engine or from a local/gdrive
+  depending on parameters
+  --------------------------------------------------
+  Parameters:
+  required_rasters:
+    The list of rasters to load, e.g. ["DEM", "PA", "VPD"]
+  use_earth_engine_assets: 
+    Whether to load the rasters from earth engine. If False, loads from local/gdrive.
+  local_fallback:
+    If use_earth_engine_assets=True, whether to fallback to gdrive if rasters could not
+    be found in ee. 
+  """
+
+  rasters_to_dispatch = {}
+
+  if use_earth_engine_assets:
+    for raster_name in required_rasters:
+      if raster_name in eeraster.column_name_to_ee_asset_fn:
+        rasters_to_dispatch[raster_name] = eeraster.column_name_to_ee_asset_fn[raster_name]()
+
+  if not use_earth_engine_assets or local_fallback:
+    for raster_name in required_rasters:
+      
+      # Skip loading locally if it was found in EE.
+      if raster_name in rasters_to_dispatch:
+        continue
+      
+      # Load it from local (or gdrive).
+      if raster_name in raster.column_name_to_geotiff_fn:
+        rasters_to_dispatch[raster_name] = raster.column_name_to_geotiff_fn[raster_name]()
+
+  # Identify missing rasters.
+  missing = set(rasters_to_dispatch.keys()) -  set(required_rasters)
+  if missing:
+    raise ValueError("The following training columns do not have an associated raster: ", missing)
+
+  return rasters_to_dispatch
+
+  
+
+def generate_isoscapes_from_variational_model(
+    model: model.Model,
+    res_x: int, 
+    res_y: int,
+    output_geotiff: str,
+    amazon_only: bool=False):
+  """
+  generate_isoscapes_from_variational_model function
+  --------------------------------------------------
+  This function generates an isoscape using a model, according
+  to the resolution specs. It queries the model for every 
+  pixel in a (res_x x res_y) tiff of the landscape.
+  --------------------------------------------------
+  Parameters:
+  model: model.Model 
+    Pretrained model used to make predictions on every pixel.
+  res_x: int
+    The output x resolution
+  res_y: int
+    The output y resolution
+  output_geotiff: str
+    Name of the file to output. 
+  amazon_only: bool
+    Whether to only generate a raster of the Amazon region as opposed to
+    all of Brazil.
+  """
+  required_geotiffs = model.training_column_names()
+  required_geotiffs.remove('lat')
+  required_geotiffs.remove('long')
+  
+  input_geotiffs = dispatch_rasters(
+    required_geotiffs,
+    use_earth_engine_assets=True,
+    local_fallback=True)
+
+  arbitrary_geotiff = raster.dem_geotiff()
+  if amazon_only:
+    arbitrary_geotiff = raster.d13C_mean_amazon_only_geotiff()
+  base_bounds = raster.get_extent(arbitrary_geotiff.gdal_dataset)
+  output_resolution = raster.create_bounds_from_res(res_x, res_y, base_bounds) 
+
+  np = get_predictions_at_each_pixel(
```

Comment:
Using the `np` name for a local variable violates the widely-held convention of reserving `np` for the numpy import. Please change the name of this variable.